src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,184 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> first;
+    private Node<T> last;
+
     @Override
     public void add(T value) {
+        if (isEmpty()) {
+            setFirstValue(value);
+            return;
+        }
+        linkLast(value);
     }
 
     @Override
     public void add(T value, int index) {
+        checkAddIndex(index);
+        if (isEmpty()) {
+            setFirstValue(value);
+            return;
+        }
+        if (index == size) {
+            linkLast(value);
+            return;
+        }
+        linkBefore(value, findNodeByIndex(index));
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (int i = 0; i < list.size(); i++) {
+            this.add(list.get(i));
+        }
     }
```

Comment:
no need to call two methods when add all, just call the add method for each List value